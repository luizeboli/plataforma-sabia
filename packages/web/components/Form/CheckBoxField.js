import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledCheckBox = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	padding: 1rem;
	font: 1.2rem;
`;

const StyledCheckBoxInput = styled.input`
	width: 2.5rem;
	height: 2.5rem;
	margin: 0rem;
	margin: 0 0.5rem 0 0;
`;
const StyledCheckBoxLabel = styled.div`
	width: 100%;
	color: ${({ theme }) => theme.colors.lightGray};
`;

const CheckBoxField = ({ name, value, label, required, onChange }) => {
	const handleOnChange = () => onChange((oldValue) => !oldValue);
	return (
		<StyledCheckBox>
			<StyledCheckBoxInput
				id={name}
				name={name}
				type="checkbox"
				aria-label={label}
				aria-required={required}
				required={required}
				checked={value}
				onChange={handleOnChange}
			/>
			<StyledCheckBoxLabel>{label}</StyledCheckBoxLabel>
		</StyledCheckBox>
	);
};
CheckBoxField.propTypes = {
	name: PropTypes.string.isRequired,
	value: PropTypes.bool,
	label: PropTypes.string,
	required: PropTypes.bool,
	onChange: PropTypes.func,
};

CheckBoxField.defaultProps = {
	value: false,
	label: '',
	required: false,
	onChange: () => {},
};

export default CheckBoxField;
