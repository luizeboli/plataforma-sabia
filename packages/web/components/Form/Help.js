import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { AiFillQuestionCircle } from 'react-icons/ai';
import ReactTooltip from 'react-tooltip';

const Icon = styled(AiFillQuestionCircle)`
	width: 4rem;
	height: 4rem;
	margin: 0.6rem 0.5rem;
	fill: ${({ theme }) => theme.colors.mediumGray};
	transition: fill 0.2s;
	:hover {
		cursor: pointer;
		fill: ${({ theme }) => theme.colors.darkGray};
	}
`;

const Help = ({ id, HelpComponent }) => {
	return (
		<>
			<Icon data-tip data-for={id} />
			<ReactTooltip uuid="uuid" id={id} multiline type="dark" place="bottom" effect="solid">
				{HelpComponent}
			</ReactTooltip>
		</>
	);
};

Help.propTypes = {
	id: PropTypes.string,
	HelpComponent: PropTypes.node,
};

Help.defaultProps = {
	id: '',
	HelpComponent: <p>Help Text</p>,
};

export default Help;
