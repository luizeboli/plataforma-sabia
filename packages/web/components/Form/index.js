export { default as InputField } from './InputField';
export { default as SwitchField } from './SwitchField';
export { default as TextField } from './TextField';
export { default as SelectField } from './SelectField';
export { default as CheckBoxField } from './CheckBoxField';
export { default as MaskedInputField } from './MaskedInputField';
export { default as Watcher } from './Watcher';
export { default as InputHiddenField } from './InputHiddenField';
export { default as Repeater } from './Repeater';
export { default as Help } from './Help';
export * from './Form';
