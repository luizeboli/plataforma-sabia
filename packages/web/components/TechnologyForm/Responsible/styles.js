import styled from 'styled-components';

export const Row = styled.div`
	display: flex;
	align-items: center;

	& > *:not(:first-child):not(:last-child) {
		margin: 0 10px;
	}

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		flex-direction: column;
		margin-top: 1.5rem;
	}
`;

export const Col = styled.div`
	flex: ${({ size }) => size || 1};

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		width: 100%;
	}
`;

export const Wrapper = styled.div`
	margin-bottom: 4rem;
`;
