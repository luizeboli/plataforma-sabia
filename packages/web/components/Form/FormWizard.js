import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { AiOutlineCheck } from 'react-icons/ai';
import { Form, Actions } from './Form';
import { Button } from '../Button';

const FormWizardContainer = styled.div``;

const StepsContainer = styled.div`
	width: 100%;
	background: ${({ theme }) => theme.colors.primary};
	border-top: 4px solid ${({ theme }) => theme.colors.lightGray};
	padding: 3rem 0 5rem 0;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		padding: 1rem 0 4rem 0;
	}
`;

const MobileSteps = styled.ol`
	display: flex;
	flex-direction: column;
	align-items: center;

	div {
		display: flex;
		justify-content: center;
		align-items: center;
		list-style: none;

		> li:nth-child(2) {
			font-size: 6rem;
		}

		> li:not(:last-child) {
			margin-right: 2rem;
		}
	}

	@media (min-width: ${({ theme }) => theme.screens.medium}px) {
		display: none;
	}
`;

const WebSteps = styled.ol`
	display: flex;
	justify-content: center;
	align-items: center;
	list-style: none;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		display: none;
	}
`;

const StepItem = styled.li`
	position: relative;
	display: flex;
	align-items: center;

	${({ completed }) =>
		completed &&
		css`
			> div {
				> p {
					color: ${({ theme }) => theme.colors.darkGray};
				}
				> span {
					color: ${({ theme }) => theme.colors.white};
					background: ${({ theme }) => theme.colors.darkGray};
				}
			}
		`};

	> div {
		display: flex;
		align-items: center;
		flex-direction: column;

		> p {
			height: 2rem;
		}
	}

	::after {
		content: '';
		display: block;
		position: relative;
		width: 100px;
		height: 12px;
		box-sizing: content-box;
		outline: 1px solid
			${({ theme, completed }) => (completed ? theme.colors.darkGray : theme.colors.white)};
		background: ${({ theme, completed }) =>
			completed ? theme.colors.darkGray : theme.colors.white};
	}

	:last-child ::after {
		display: none;
	}

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		::after {
			display: none;
		}
	}
`;

const StepLabel = styled.p`
	font-size: 1.4rem;
	color: ${({ theme }) => theme.colors.white};
	font-weight: 700;
	position: absolute;
	bottom: -25px;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		position: relative;
		color: ${({ theme }) => theme.colors.darkGray};
	}
`;

const StepNumber = styled.span`
	background: ${({ theme }) => theme.colors.white};
	display: block;
	text-align: center;
	font-size: 4.5rem;
	height: 6rem;
	width: 6rem;
	line-height: 6rem;
	border-radius: 50%;
	position: relative;

	> svg {
		height: 100%;
	}
`;

const FormWizard = ({ steps, currentStep, onSubmit, onPrev }) => {
	const CurrentFormStep =
		currentStep !== '' ? steps.find((step) => step.slug === currentStep).form : steps[0].form;

	const currentStepSlug = currentStep || steps[0].slug;
	let currentStepIndex = 0;
	steps.forEach((step, i) => {
		if (step.slug === currentStepSlug) {
			currentStepIndex = i;
		}
	});

	const nextStep =
		currentStepIndex === steps.length - 1 ? false : steps[currentStepIndex + 1].slug;
	const prevStep = currentStepIndex === 0 ? false : steps[currentStepIndex - 1].slug;

	const handleSubmit = (data) => {
		window.scrollTo({ top: 0 });
		onSubmit({ data, step: currentStepSlug, nextStep });
	};

	const handlePrev = () => {
		window.scrollTo({ top: 0 });
		onPrev({ step: currentStepSlug, prevStep });
	};

	return (
		<FormWizardContainer>
			<StepsContainer>
				<WebSteps>
					{steps.map((step, i) => {
						const showIcon = i < currentStepIndex || typeof step.icon !== 'undefined';
						const Icon = i < currentStepIndex ? AiOutlineCheck : step.icon || null;
						return (
							<StepItem completed={i <= currentStepIndex} key={step.slug}>
								<div>
									<StepNumber>{showIcon ? <Icon /> : i + 1}</StepNumber>
									<StepLabel>{step.label}</StepLabel>
								</div>
							</StepItem>
						);
					})}
				</WebSteps>
				<MobileSteps>
					<div>
						<StepItem completed>
							<div>
								<StepNumber>{currentStepIndex + 1}</StepNumber>
							</div>
						</StepItem>
						<li>/</li>
						<StepItem>
							<div>
								<StepNumber>{steps.length}</StepNumber>
							</div>
						</StepItem>
					</div>
					<StepLabel>{steps[currentStepIndex].label}</StepLabel>
				</MobileSteps>
			</StepsContainer>

			<Form onSubmit={handleSubmit}>
				{CurrentFormStep && <CurrentFormStep />}
				<Actions center>
					{prevStep && (
						<Button variant="secondary" onClick={handlePrev}>
							Voltar
						</Button>
					)}
					<Button type="submit">Salvar e Continuar</Button>
				</Actions>
			</Form>
		</FormWizardContainer>
	);
};

FormWizard.propTypes = {
	onSubmit: PropTypes.func,
	onPrev: PropTypes.func,
	steps: PropTypes.arrayOf(
		PropTypes.shape({
			label: PropTypes.string.isRequired,
			slug: PropTypes.string.isRequired,
			form: PropTypes.elementType,
			icon: PropTypes.elementType,
		}),
	).isRequired,
	currentStep: PropTypes.string.isRequired,
};

FormWizard.defaultProps = {
	onSubmit: () => {},
	onPrev: () => {},
};

export default FormWizard;
