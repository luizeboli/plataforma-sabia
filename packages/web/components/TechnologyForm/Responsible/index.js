/* eslint-disable no-use-before-define */
import React from 'react';
import PropTypes from 'prop-types';
import { FaPlus, FaMinus } from 'react-icons/fa';
import { InputField } from '../../Form';
import { Col, Row, Wrapper } from './styles';
import { CircularButton } from '../../Button';
import Repeater from '../../Form/Repeater';

const Responsible = ({ form }) => {
	const emptyValue = {
		fullName: '',
		email: '',
		phoneNumber: '',
		lattesID: '',
	};

	return (
		<Wrapper>
			<Repeater
				form={form}
				name="responsible"
				emptyValue={emptyValue}
				childsComponent={({ item, index, remove, fields }) => {
					return (
						<>
							<Row key={item.id}>
								<Col size={2}>
									<InputField
										form={form}
										name={`responsible[${index}].fullName`}
										label="Nome Completo"
										placeholder="Nome do responsável"
										validation={{ required: true }}
									/>
								</Col>
								<Col>
									<InputField
										form={form}
										name={`responsible[${index}].email`}
										label="Email"
										placeholder="Ex.: email@dominio.com.br"
										validation={{ required: true }}
									/>
								</Col>
								<Col>
									<InputField
										form={form}
										name={`responsible[${index}].phone`}
										label="Telefone"
										placeholder="(xx) xxxx - xxxx"
										validation={{ required: true }}
									/>
								</Col>
								<Col>
									<InputField
										form={form}
										name={`responsible[${index}].lattesId`}
										label="ID Lattes"
										placeholder="Somente números"
										validation={{ required: true }}
									/>
								</Col>

								<CircularButton
									disabled={index === 0 && fields.length === 1}
									small
									variant="remove"
									shortPadding
									onClick={(event) => {
										event.preventDefault();
										remove(index);
									}}
								>
									<FaMinus />
								</CircularButton>
							</Row>
						</>
					);
				}}
				// eslint-disable-next-line no-shadow
				endComponent={({ append, emptyValue }) => {
					return (
						<CircularButton
							right
							variant="info"
							color="white"
							onClick={(event) => {
								event.preventDefault();
								append(emptyValue);
							}}
						>
							<FaPlus />
						</CircularButton>
					);
				}}
			/>
		</Wrapper>
	);
};

Responsible.propTypes = {
	form: PropTypes.shape({}),
};

Responsible.defaultProps = {
	form: {},
};

export default Responsible;
