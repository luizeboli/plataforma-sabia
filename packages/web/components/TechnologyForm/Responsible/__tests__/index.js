/* eslint-disable jest/no-disabled-tests */
import React from 'react';
import { render, fireEvent, screen } from 'test-utils';
import Responsible from '..';
import { Form } from '../../../Form';

const onSubmit = jest.fn(() => {});

jest.mock('react-icons/fa', () => {
	return {
		__esModule: true,
		FaPlus: () => {
			return <svg>FaPlus mocked</svg>;
		},
		FaMinus: () => {
			return <svg>FaMinus mocked</svg>;
		},
	};
});

test('it renders the form with one row', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<Responsible />
		</Form>,
	);

	expect(container).toMatchSnapshot();
});

test('it increase the number of rows when clicking in add button', () => {
	render(
		<Form onSubmit={onSubmit}>
			<Responsible />
		</Form>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	expect(buttons[1]).toMatchSnapshot();
});

test('it decrease the number of rows when clicking in remove button', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<Responsible />,
		</Form>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	const buttons2 = screen.getAllByRole('button');

	fireEvent.click(buttons2[0]);

	expect(container).toMatchSnapshot();
});

test('it should not remove the row when only one is available', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<Responsible />
		</Form>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[0]);

	expect(container).toMatchSnapshot();
});
