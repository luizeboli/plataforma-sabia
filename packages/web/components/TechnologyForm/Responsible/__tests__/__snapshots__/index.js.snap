// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it decrease the number of rows when clicking in remove button 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c8 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c11 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c9 {
    width: 100%;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          size="2"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].fullName"
            >
              Nome Completo
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Nome Completo"
                aria-required="true"
                class="c7"
                id="responsible[0].fullName"
                name="responsible[0].fullName"
                placeholder="Nome do responsável"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].email"
            >
              Email
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Email"
                aria-required="true"
                class="c7"
                id="responsible[0].email"
                name="responsible[0].email"
                placeholder="Ex.: email@dominio.com.br"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].phone"
            >
              Telefone
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Telefone"
                aria-required="true"
                class="c7"
                id="responsible[0].phone"
                name="responsible[0].phone"
                placeholder="(xx) xxxx - xxxx"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].lattesId"
            >
              ID Lattes
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="ID Lattes"
                aria-required="true"
                class="c7"
                id="responsible[0].lattesId"
                name="responsible[0].lattesId"
                placeholder="Somente números"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          disabled=""
          title=""
          type="button"
        >
          <svg>
            FaMinus mocked
          </svg>
        </button>
      </div>
      <button
        class="c11"
        color="hsla(0, 0%, 100%)"
        title=""
        type="button"
      >
        <svg>
          FaPlus mocked
        </svg>
      </button>
    </div>
    ,
  </form>
</div>
`;

exports[`it increase the number of rows when clicking in add button 1`] = `
.c0 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0:hover {
  opacity: 0.8;
}

@media (max-width:450px) {

}

@media (max-width:450px) {

}

@media (max-width:450px) {

}

<button
  class="c0"
  color="hsla(0, 0%, 100%)"
  title=""
  type="button"
>
  <svg>
    FaPlus mocked
  </svg>
</button>
`;

exports[`it renders the form with one row 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c8 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c11 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c9 {
    width: 100%;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          size="2"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].fullName"
            >
              Nome Completo
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Nome Completo"
                aria-required="true"
                class="c7"
                id="responsible[0].fullName"
                name="responsible[0].fullName"
                placeholder="Nome do responsável"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].email"
            >
              Email
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Email"
                aria-required="true"
                class="c7"
                id="responsible[0].email"
                name="responsible[0].email"
                placeholder="Ex.: email@dominio.com.br"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].phone"
            >
              Telefone
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Telefone"
                aria-required="true"
                class="c7"
                id="responsible[0].phone"
                name="responsible[0].phone"
                placeholder="(xx) xxxx - xxxx"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].lattesId"
            >
              ID Lattes
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="ID Lattes"
                aria-required="true"
                class="c7"
                id="responsible[0].lattesId"
                name="responsible[0].lattesId"
                placeholder="Somente números"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          disabled=""
          title=""
          type="button"
        >
          <svg>
            FaMinus mocked
          </svg>
        </button>
      </div>
      <button
        class="c11"
        color="hsla(0, 0%, 100%)"
        title=""
        type="button"
      >
        <svg>
          FaPlus mocked
        </svg>
      </button>
    </div>
  </form>
</div>
`;

exports[`it should not remove the row when only one is available 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c8 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c11 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c9 {
    width: 100%;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          size="2"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].fullName"
            >
              Nome Completo
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Nome Completo"
                aria-required="true"
                class="c7"
                id="responsible[0].fullName"
                name="responsible[0].fullName"
                placeholder="Nome do responsável"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].email"
            >
              Email
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Email"
                aria-required="true"
                class="c7"
                id="responsible[0].email"
                name="responsible[0].email"
                placeholder="Ex.: email@dominio.com.br"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].phone"
            >
              Telefone
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="Telefone"
                aria-required="true"
                class="c7"
                id="responsible[0].phone"
                name="responsible[0].phone"
                placeholder="(xx) xxxx - xxxx"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="responsible[0].lattesId"
            >
              ID Lattes
            </label>
            <div
              class="c6"
            >
              <input
                aria-label="ID Lattes"
                aria-required="true"
                class="c7"
                id="responsible[0].lattesId"
                name="responsible[0].lattesId"
                placeholder="Somente números"
                type="text"
              />
            </div>
            <span
              class="c8"
            />
          </div>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          disabled=""
          title=""
          type="button"
        >
          <svg>
            FaMinus mocked
          </svg>
        </button>
      </div>
      <button
        class="c11"
        color="hsla(0, 0%, 100%)"
        title=""
        type="button"
      >
        <svg>
          FaPlus mocked
        </svg>
      </button>
    </div>
  </form>
</div>
`;
